### PR TITLE
Update proc_creating_custom_test.adoc

### PR DESCRIPTION
**Pull request by jinqi7**
Add description about the Params/Labels automatically added by integration-service

### CODE REVIEW

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc [hunk 1]
```diff
@@ -91,6 +91,16 @@ spec:
...
 .. For additional instructions on adding an integration test, see this document: xref:how-to-guides/testing_applications/proc_adding_an_integration_test.adoc[Adding an integration test].
 
+----
+In the integration test created above, integration-service adds some Params/Labels automatically to the integration PipelineRun
+
+1. A param containing the Snapshot as a json string
+2. A param containing the DeploymentTarget connection details and Environment name
+3. A Label for type of PipelineRun, i.e. "test"
+4. A Label for optional flag of IntegrationTestScenario 
+5. Labels for the name of application, component and IntegrationTestScenario
```
Comment (MartinBasti) [inline docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc:101]: Shall we write explicitly names of params and labels?

Also examples would be helpful

**Comment by MartinBasti** [comment]
There are 2 commits with the same name, they should be squashed probably into one.

Also commit message should describe what is new, not just which file is being updated

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc (hunk 1 above)
Comment (jinqi7) [reply docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc:101]: What examples? Do you mean the current example for the IntegrationPipelineRun is too simple and replace it with a new one?

**Comment by jinqi7** [comment]
> There are 2 commits with the same name, they should be squashed probably into one.
> 
> Also commit message should describe what is new, not just which file is being updated

Since this PR is not generated/ updated in our normal way, I'll try to squash it or I'll create a new PR.

**Comment by MartinBasti** [comment]
> Since this PR is not generated/ updated in our normal way, I'll try to squash it or I'll create a new PR.

Can you elaborate more?

**Comment by jinqi7** [comment]
> > Since this PR is not generated/ updated in our normal way, I'll try to squash it or I'll create a new PR.
> 
> Can you elaborate more?

Sure. Squashed.

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc (hunk 1 above)
Comment (MartinBasti) [reply docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc]: yes, like complete value what user can see in SNAPSHOT param

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc [hunk 2]
```diff
@@ -91,6 +91,18 @@ spec:
 . Add your new custom test as an integration test in {ProductName}.
 .. For additional instructions on adding an integration test, see this document: xref:how-to-guides/testing_applications/proc_adding_an_integration_test.adoc[Adding an integration test].
 
+----
```
Comment (MartinBasti) [inline docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc:94]: Why is added as a code block and not standard text?

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc [hunk 3]
```diff
@@ -91,6 +91,18 @@ spec:
...
+In the integration test created above, integration-service adds some Params/Labels automatically to the integration PipelineRun
+
+1. The param of "SNAPSHOT" containing the Snapshot as a json string
+2. The param of "NAMESPACE" containing the namespace where the snapshot will be deployed 
+3. The Label of "appstudio.openshift.io/snapshot" containing the name of snapshot
+4. The Label of "test.appstudio.openshift.io/optional" for optional flag of IntegrationTestScenario
+5. The Label of "test.appstudio.openshift.io/scenario" containing the name of IntegrationTestScenario
+6. The Label of "appstudio.openshift.io/application" containing the name of application
+7. The Label of "appstudio.openshift.io/component" containing the name of component
```
Comment (MartinBasti) [inline docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc:103]:
```suggestion
.Data injected into integration PipelineRun

In the integration test created above, integration-service adds following items automatically to the integration PipelineRun.

Parameters:

. `SNAPSHOT` containing the Snapshot as a json string
. `NAMESPACE` containing the namespace where the snapshot will be deployed 

Labels:
. `appstudio.openshift.io/application` containing the name of application
. `appstudio.openshift.io/component` containing the name of component
. `appstudio.openshift.io/snapshot` containing the name of snapshot
. `test.appstudio.openshift.io/optional` for optional flag of IntegrationTestScenario
. `test.appstudio.openshift.io/scenario` containing the name of IntegrationTestScenario
```

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc (hunk 2 above)
Comment (jinqi7) [reply docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc]: This part is not a step of the procedure. I think it should be a additional Note or Tips. I updated to use Note in the PR.

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc (hunk 1 above)
Comment (jinqi7) [reply docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc]: @MartinBasti , do you mean add an example value for SNAPSHOT like this one ? {"application":"application-sample","components":[{"name":"component-sample","containerImage":"quay.io/redhat-appstudio/sample-image@sha256:34fb1c72bc976937d0a59849160746cec12a8c925f31e74dc9f3b922f74da0e9","source":{"git":{"url":"git@github.com:redhat-appstudio/integration-service.git\n","revision":"sha256:3703843e80f39188c060a6ed0f524519a682be1e7141fee29b878b1973e9783c"}}}],"artifacts":{}}

Comment (jinqi7) [reply docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc]: It's so long that I don't want to add it.

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc (hunk 3 above)
Comment (jinqi7) [reply docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc]: The format is not working well with the suggested change.

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc [hunk 4]
```diff
@@ -29,68 +29,75 @@ metadata:
...
+            value: tasks/task-1.yaml
       params:
-      - name: username
-      steps:
-      - image: alpine:3.15
-        script: |
-          echo "hello $(params.username)"
-
+        - name: username
+          value: $(params.username)
```
Comment (MartinBasti) [inline docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc:47]: Why this change? 
If you want to fix unrelated parts to injected variables, open a new PR.

I discontinue review, I'm not going to review PR with changing scope

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc [hunk 5]
```diff
@@ -29,68 +29,75 @@ metadata:
...
+.. If you haven’t already, commit the files to the GitHub repository that {ProductName} can access.
+
+
 . Add your new custom test as an integration test in {ProductName}.
 .. For additional instructions on adding an integration test, see this document: xref:how-to-guides/testing_applications/proc_adding_an_integration_test.adoc[Adding an integration test].
 
+
+[NOTE]
+====
+In the integration test created above, integration-service adds some Params/Labels automatically to the integration PipelineRun.
```
Comment (MartinBasti) [inline docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc:85]:
```suggestion
In the integration test created above, integration-service adds following Parameters/Labels automatically to the integration PipelineRun. 
```

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc (hunk 1 above)
Comment (MartinBasti) [reply docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc]: yes

that's why I wanted to make it a separate textual section, which gives us more space

if SNAPSHOT JSON cannot be added, we should at least link it to SNAPSHOT definition, so use can easily find which fields are available

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc (hunk 4 above)
Comment (jinqi7) [reply docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc:47]: It's mentioned in the A/C item -  Examples and usage for docs team (file a story), i.e. how user can create working integration test to test deployed services are provided. So you meant to file a new JIRA ticket for it ?

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc (hunk 1 above)
Comment (jinqi7) [reply docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc]: Added the link to Snapshot definition as you suggested. Thanks

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc (hunk 4 above)
Comment (MartinBasti) [reply docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc]: But using those options we are documenting, SNAPSHOT param

if all examples are completelly wrong, then we should have a new ticket

Comment (MartinBasti) [reply docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc]: what I want to achieve here is, that it's clear to users how to use those parameters for creating their own integration test tasks:
- user should be able to find in doc what is injected into test pipeline by appstudio/integration-service
- user should be able to clearly understand what the values are (mainly in structured data like SNAPSHOT)
- user should be able to understand how to easily use params, and how to integrate them into testing pipeline (one example is better than 1000 words)

So maybe we can have a subsection here, with just single example, how to construct task which can use SNAPSHOT (for example, task may just print image url)

Comment (jinqi7) [reply docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc]: An example needs its context. With the original pipeline and task in the page, it's hard to add a "simple example" for the SNAPSHOT from my understand.

Comment (MartinBasti) [reply docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc]: ok then

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc (hunk 5 above)
Comment (jinqi7) [reply docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc]: Updated. Thanks

**Comment by jinqi7** [comment]
@dirgim, can you please help to review the PR? Thanks

**Comment by jinqi7** [comment]
@MartinBasti , can you please mark your comments as resolved if you think they are?

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc [hunk 6]
```diff
@@ -91,6 +91,23 @@ spec:
...
 .. For additional instructions on adding an integration test, see this document: xref:how-to-guides/testing_applications/proc_adding_an_integration_test.adoc[Adding an integration test].
 
+.Data injected into integration PipelineRun
+
+In the integration test created above, integration-service adds the following Parameters/Labels automatically to the integration PipelineRun.
+
+Parameters:
+
+ . `SNAPSHOT` containing the https://github.com/redhat-appstudio/application-api/blob/main/api/v1alpha1/snapshot_types.go[Snapshot] as a json string
+ . `NAMESPACE` containing the namespace where the snapshot will be deployed
```
Comment (dirgim) [inline docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc:101]: Can you clarify here that this parameter only gets added if the IntegrationTestScenario uses ephemeral environments to deploy the Snapshot that's being tested?

Comment (jinqi7) [reply docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc:101]: Thanks. Added a NOTE for it.

Comment (dirgim) [reply docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc:101]: Thanks, looks good

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc [hunk 7]
```diff
@@ -91,6 +91,28 @@ spec:
 . Add your new custom test as an integration test in {ProductName}.
 .. For additional instructions on adding an integration test, see this document: xref:how-to-guides/testing_applications/proc_adding_an_integration_test.adoc[Adding an integration test].
 
+.Data injected into integration PipelineRun
+
+In the integration test created above, integration-service adds the following Parameters/Labels automatically to the integration PipelineRun.
+
+Parameters:
+
+ . `SNAPSHOT` containing the https://github.com/redhat-appstudio/application-api/blob/main/api/v1alpha1/snapshot_types.go[Snapshot] as a json string
```
Comment (dirgim) [inline docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc:100]: Just an idea: maybe we can have the example contents of the Snapshot stored in our [integration-examples](https://github.com/redhat-appstudio/integration-examples) repository and link to it here to provide the users with a better idea of how it will look.

Comment (jinqi7) [reply docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc:100]: Do you mean we store the example content of the Snapshot [here](https://github.com/redhat-appstudio/integration-examples/blob/main/pipelines/integration_test_app.yaml#L13C3-L13C3)? I just wonder if we need to change the pipeline/task in the page also?

Comment (dirgim) [reply docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc:100]: We could have a new directory called `examples` with a more well defined example of the Snapshot like the one you mentioned earlier:
```
{"application":"application-sample","components":[{"name":"component-sample","containerImage":"quay.io/redhat-appstudio/sample-image@sha256:34fb1c72bc976937d0a59849160746cec12a8c925f31e74dc9f3b922f74da0e9","source":{"git":{"url":"[git@github.com](mailto:git@github.com):redhat-appstudio/integration-service.git\n","revision":"sha256:3703843e80f39188c060a6ed0f524519a682be1e7141fee29b878b1973e9783c"}}
```

Comment (jinqi7) [reply docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc:100]: @dirgim, so, I just need to create a new file contains the example content of Snapshot , right?

Comment (dirgim) [reply docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc:100]: Yes, I think that would do the job.

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc [hunk 8]
```diff
@@ -91,6 +91,28 @@ spec:
...
+====
+The NAMESPACE parameter only gets added if the IntegrationTestScenario uses ephemeral environments to deploy the Snapshot that's being tested
+====
+
+Labels:
+
+ . `appstudio.openshift.io/application` containing the name of application
+ . `appstudio.openshift.io/component` containing the name of component
+ .  `appstudio.openshift.io/snapshot` containing the name of snapshot
+ .  `test.appstudio.openshift.io/optional` for optional flag of IntegrationTestScenario
```
Comment (MartinBasti) [inline docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc:113]:
```suggestion
 . `appstudio.openshift.io/snapshot` containing the name of snapshot
 . `test.appstudio.openshift.io/optional` for optional flag of IntegrationTestScenario
```

Comment (jinqi7) [reply docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_creating_custom_test.adoc:113]: Resolved

**Comment by jinqi7** [comment]
> > @MartinBasti , can you please mark your comments as resolved if you think they are?
> 
> I cannot, I'm don't have permissions.
> 
> Also, we still don't have example how to use SNAPSHOT param in a task

If a task is also needed, how about just use this [one](https://github.com/redhat-appstudio/integration-examples/blob/main/tasks/test_app_endpoint.yaml#L39C8-L39C74)?

**Comment by MartinBasti** [comment]
> > > @MartinBasti , can you please mark your comments as resolved if you think they are?
> > 
> > 
> > I cannot, I'm don't have permissions.
> > Also, we still don't have example how to use SNAPSHOT param in a task
> 
> If a task is also needed, how about just use this [one](https://github.com/redhat-appstudio/integration-examples/blob/main/tasks/test_app_endpoint.yaml#L39C8-L39C74)? Or, an new task like [this](https://github.com/redhat-appstudio/integration-examples/blob/2b02112f60ceb67910c7420641cdfb1cc2d250cd/examples/task.yaml) ? Is a new pipeline file needed als?@MartinBasti @dirgim

that works for me

**Comment by Chr1st1anSears** [comment]
Hey @jinqi7--thanks for your PR! I'll be creating a separate PR, which is how we'll actually merge your content, but rest assured, it will contain your requested changes. 
Could you please answer a few questions?
1. What exactly is the `integration-service`? 
2. In a few words, how would you describe the purpose of the data that gets injected into the integration PipelineRun? How do these parameters and labels help users?
3. Where should your second link in the line about SNAPSHOT point to? Currently, when I click on "An example of Snapshot json string" I get an error.

**Comment by jinqi7** [comment]
> Hey @jinqi7--thanks for your PR! I'll be creating a separate PR, which is how we'll actually merge your content, but rest assured, it will contain your requested changes. Could you please answer a few questions?
> 
> 1. What exactly is the `integration-service`? 
It's just the "integration service". 
> 2. In a few words, how would you describe the purpose of the data that gets injected into the integration PipelineRun? How do these parameters and labels help users?
 The paramters  are passed into the pipelineRun as input. The labels can help user to search the pipelineRun.
> 3. Where should your second link in the line about SNAPSHOT point to? Currently, when I click on "An example of Snapshot json string" I get an error. 
You'll see it when the [PR](https://github.com/redhat-appstudio/integration-examples/pull/12) is merged.

**Comment by jinqi7** [comment]
> Hey @jinqi7--thanks for your PR! I'll be creating a separate PR, which is how we'll actually merge your content, but rest assured, it will contain your requested changes. Could you please answer a few questions?
> 
@Chr1st1anSears , I can't understand why a separte PR is needed.  If you want to do some adjustment of the format, I think perhaps it can be done when the task [HACDOCS-519 ](https://issues.redhat.com/browse/HACDOCS-519)is being done. Wdyt?

**Comment by Chr1st1anSears** [comment]
@jinqi7 as you can see in [this document](https://docs.google.com/document/d/18uq_RWxAdBgEsYsPenOre89Q_v6YJp3wvQ8aiXoXrwQ/edit#heading=h.i1cll0oo5uhd), our team has determined that, to accept content from non-writers, writers will create a separate PR with that new content.
Also, it would help if you answered my questions a little more thoroughly. Technically speaking, what is the integration service? What exactly does it do? And yes parameters are inputs, but what purpose do these parameters serve? Why exactly would users want to know about these parameters, what is the use case? And I'll need the working link to the example in order to include it.

**Comment by dirgim** [comment]
Hi @Chr1st1anSears I'll share some more context on your questions:

1. The Integration Service faciliates automated testing of content produced by the build pipelines. It is mainly responsible for integration testing capabilities. It watches for Component builds, triggers and manages testing Tekton Pipelines, and creates releases based on the testing outcome(s). You can read more in the [RHTAP book page on it](https://redhat-appstudio.github.io/book/book/integration-service.html) and feel free to reach out to the team's #forum-rhtap-integration on slack.
2. Here are quick blurbs about the individual parameters:
    - The `SNAPSHOT` parameter is a JSON representation of the application `Snapshot` resource that's under test. It provides the user with the information on which exact components and their images are being tested, as well as additional information such as the originating git repository and commit that was used to build individual components. 
    - The `NAMESPACE` parameter is used only for the kind of Integration testing which deploys the Snapshot under test to ephemeral testing environments and is used to indicate which namespace is being used for testing. This is intended to help the user when querying for kubernetes resources that are deployed in the test environment, such as routes, deployments, pods, and similar.
3. The example JSON from the [mentioned URL](https://github.com/redhat-appstudio/integration-examples/blob/main/examples/snapshot_json_string_example) is now available.

**Comment by Chr1st1anSears** [comment]
Closing this PR per docs team process. Please see the new PR that will get merged to add this content: https://github.com/redhat-appstudio/docs.appstudio.io/pull/185

**Comment by MartinBasti** [comment]
> @jinqi7 as you can see in [this document](https://docs.google.com/document/d/18uq_RWxAdBgEsYsPenOre89Q_v6YJp3wvQ8aiXoXrwQ/edit#heading=h.i1cll0oo5uhd), our team has determined that, to accept content from non-writers, writers will create a separate PR with that new content. Also, it would help if you answered my questions a little more thoroughly. Technically speaking, what is the integration service? What exactly does it do? And yes parameters are inputs, but what purpose do these parameters serve? Why exactly would users want to know about these parameters, what is the use case? And I'll need the working link to the example in order to include it.

Would be nice to have it mentioned in https://github.com/redhat-appstudio/docs.appstudio.io/blob/main/CONTRIBUTING.md for contributors instead of a hidden google doc
